configure logging for buildbot-worker runner

rutsky opened the pull request:
As alternative we can revert changes from 4d733e6 commit by @jvlomax
(in buildbot-worker copy).

By default Twisted log is not written anywhere so we don't see any
errors from buildbot-worker invocations, e.g.

```
$ buildbot-worker create-worker w
$ echo $?
1
$
```

tardyp commented:
LGTM

tardyp commented:
twisted.logger does not exist in twisted 12..

sa2ajj commented on worker/buildbot_worker/scripts/runner.py:
```diff
@@ -20,9 +20,13 @@
 from twisted.python import log
 from twisted.python import reflect
 from twisted.python import usage
+from zope.interface import implementer
```
should we put an empty line before this one?

rutsky replied:
@sa2ajj done

rutsky commented:
Rebased on master, I believe Twisted 12 is not a problem anymore.

No tests yet.

sa2ajj commented:
Let's see how it fares...  (I'll remove "needs work" for now though)

rutsky commented:
I resurrected `StdoutAssertionsMixin` and added lines for testing that error message ended in `stdout`.

I changed used logging API — now it works on Twisted 14.0 and 16.1 in my env.

rutsky commented:
This bug is not as easy as I expected.

Logger is currently managed in the following places:

1. When `buildbot-worker` is invoked and doesn't do anything interesting (e.g. `buildbot-worker --help`) logger is not initialized and managed (but used to output messages).

2. When `buildbot-worker` parts are used from tests, logger is initialized by test runner.

3. When worker is started by `buildbot-worker start ...` `buildbot-worker` invokes `twistd --logfile=twistd.log` log is initialized and managed by `twistd`.

Since there is no single point where logging is initialized and managed it's hard to prevent following issue: if logger is initialized by `twistd` or by test runner, it redirects all output to `stdout` to logfile, which leads to message duplication in log at best.

Also multiple initialization of logging leads to Twisted warning `Warning: primary log target selected twice`.

I think best is to revert outputting messages that are expected to be seen on command line to stdout.

rutsky commented:
I effectively reverted 4d733e6: most of `buildbot-worker` messages are now printed to stdout.

Printed messages to stdout appears in Twisted log, but not vice versa.

Having messages on stdout is critical for interactive usage of buildbot-worker
command line utility.

@jvlomax I appreciate if you comment on reason you done 4d733e6 and what do you think about my current changes.

djmitche commented:
I suspect the `log.msg` changes were made as part of Python-3-ification, and the resulting hiding of important messages wasn't noticed at the time.  So I think this revert is the right course.

rutsky commented:
@tardyp is changes still LGTY?
